### PR TITLE
Use list config for `FunctionOnlyReturningConstant>excludedFunctions`

### PR DESCRIPTION
**Pull request by BraisGabin**

### CODE REVIEW

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstantSpec.kt [hunk 1]
```diff
@@ -63,13 +63,21 @@ class FunctionOnlyReturningConstantSpec {
...
             val code = "fun f() = 1"
             val config = TestConfig(mapOf(EXCLUDED_FUNCTIONS to "f"))
             val rule = FunctionOnlyReturningConstant(config)
             assertThat(rule.compileAndLint(code)).isEmpty()
         }
 
+        @Test
+        fun `does not report excluded function which returns a constant`() {
+            val code = "fun f() = 1"
+            val config = TestConfig(mapOf(EXCLUDED_FUNCTIONS to listOf("f")))
```
Comment (marschwar) [inline detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstantSpec.kt:76]: Should this be something utilizing wildcards?

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstant.kt [hunk 2]
```diff
@@ -127,3 +127,8 @@ class FunctionOnlyReturningConstant(config: Config = Config.empty) : Rule(config
         return isConstantExpression(returnExpression?.returnedExpression)
     }
 }
+
+private operator fun Iterable<Regex>.contains(input: String?): Boolean {
+    input ?: return false
+    return any { it.matches(input) }
```
Comment (marschwar) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstant.kt:133]:
```suggestion
    return input != null && any { it.matches(input) }
```

**Review by marschwar — APPROVED** [review]

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstantSpec.kt (hunk 1 above)
Comment (BraisGabin) [reply detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstantSpec.kt:76]: I added a new test for that.

**Comment by cortinico** [comment]
I won't mention this in the notable changes as the list is getting quite long. Should I add a paragraph in the migration section of the changelog?

**Comment by cortinico** [comment]
> I won't mention this in the notable changes as the list is getting quite long. Should I add a paragraph in the migration section of the changelog?

Follow up for @arturbosch here: it seems we need this info in the changelog then 😁

**Pull request merged**